### DISCUSSION TITLE
Getting earnings release dates for stocks?

### DISCUSSION THREAD

**Discussion by Rapid1898-code** [ranaroussi/yfinance#1701] [Q&A]
Hello - i would need the earnings release dates for some stocks -

When i try output them i get this error:

```
yfinance.exceptions.YFNotImplementedError: Have not implemented fetching 'calendar' from Yahoo API
yfinance.exceptions.YFNotImplementedError: Have not implemented fetching 'earnings' from Yahoo API
```

Are there any plans to implement this functinality?
Or is this probably allready implemented in a not yet released version?

**Comment by ValueRaider** [comment]
The plan is, wait for a hero to submit an implementation.

**Comment by yunhongz** [comment]
import yfinance as yf
msft = yf.Ticker('MSFT')
msft.earnings_dates

Tried other tickers and none works

**Reply by ValueRaider** [reply]
It used to work fine. Don't spam #1932
